### DISCUSSION TITLE
model-binding-json-posts-in-asp-net-core/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#134] [Announcements]
# model-binding-json-posts-in-asp-net-core/

In this post, I am going to show what to do if you are converting a project to ASP.NET Core and you discover your JSON POSTs aren't working.

https://andrewlock.net/model-binding-json-posts-in-asp-net-core/

**Comment by rwales1000** [comment]
Thank you Andrew!  You are a blessing.

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at June 11 2016, 03:12, in reply to [khamza85](https://disqus.com/by/khamza85/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/134#discussioncomment-5794774)</em>

---

<p>Thanks for the kind words, glad you found it useful!</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at September 02 2016, 07:15, in reply to [jsgoupil](https://disqus.com/by/jsgoupil/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/134#discussioncomment-5794787)</em>

---

<p>Yeah, I feel like this is going to be an issue that pretty much anyone porting code is going to run into.</p><p>Handy tip about the wrong type mapping, thanks!</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at November 08 2016, 12:01, in reply to [Victor Cardins](https://disqus.com/by/cardins/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/134#discussioncomment-5794799)</em>

---

<p>Hi, yeah I've seen that post, I like it, if you need that behaviour then it's definitely a good option. It's not without tradeoffs though, so you have to be aware of them. In particular check out the pseudo-overposting update here <a href="https://www.billbogaiv.com/posts/updated-hybrid-model-binding-in-aspnet-core-10-rc2" rel="nofollow noopener" title="https://www.billbogaiv.com/posts/updated-hybrid-model-binding-in-aspnet-core-10-rc2">https://www.billbogaiv.com/posts/updated-hybrid-model-binding-in-aspnet-core-10-rc2</a> and here <a href="https://www.billbogaiv.com/posts/updates-to-hybrid-model-binding-in-aspnet-core-10" rel="nofollow noopener" title="https://www.billbogaiv.com/posts/updates-to-hybrid-model-binding-in-aspnet-core-10">https://www.billbogaiv.com/posts/updates-to-hybrid-model-binding-in-aspnet-core-10</a>.</p><p>I see why they made the change, and as long as you're aware of it, I think it makes sense to be explicit about what you want to bind, but it really depends on what works for you :) The biggest problem in my eyes is that it is a non-obvious breaking change you need to be aware of!</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at March 14 2017, 07:29, in reply to [Pavel Biryukov](https://disqus.com/by/pavelbiryukov/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/134#discussioncomment-5794808)</em>

---

<p>Oops, you're quite right, updated now, thanks :)</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at March 29 2017, 12:02, in reply to [Richard Bloom](https://disqus.com/by/richard_bloom/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/134#discussioncomment-5794810)</em>

---

<p>Yeah, that should work, just gave it a wizz in Postman and worked fine!</p>

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at March 29 2017, 09:37, in reply to [Richard Bloom](https://disqus.com/by/richard_bloom/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/134#discussioncomment-5794812)</em>

---

<p>No, shouldn't do, that's what I used. I suggest you post a minimal sample on Stack overflow - feel free to add the link back here if you like</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 24 2017, 08:53, in reply to [Peter Kula](https://disqus.com/by/peter_kula/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/134#discussioncomment-5794824)</em>

---

<p>Thanks for the heads-up, I've replaced the link with an equivalent post elsewhere.</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at June 15 2017, 01:30, in reply to [Peter Morris](https://disqus.com/by/disqus_UovYCEnpiZ/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/134#discussioncomment-5794828)</em>

---

<p>That's interesting, sounds like a bug, may be worth creating an issue with the ASP.NET team. I'll try and take a look myself too. Don't suppose you have a simple repro repo anywhere?</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at August 25 2020, 04:48, in reply to [Venkat Rethinam](https://disqus.com/by/venkatrethinam/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/134#discussioncomment-5794857)</em>

---

<p>Yes, you still need this. If you're using [ApiController] attribute, then it will apply the [FromBody] for you automatically to a single complex action argument. But you need to mark header source explicitly and you can only  bind to form <i>or</i> body.</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at September 01 2020, 02:33, in reply to [Bret Williams](https://disqus.com/by/disqus_DO6BRVzcgW/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/134#discussioncomment-5794860)</em>

---

<p>This post only applies to <a href="http://asp.net" rel="nofollow noopener" title="asp.net">asp.net</a> core, not <a href="http://asp.net" rel="nofollow noopener" title="asp.net">asp.net</a>.</p>

[a comment by an automated account is not shown]

**Comment by ZoltanMate** [comment]
this post helped me a lot, thank you
